network/ipfs_impl.go

```diff
@@ -312,7 +313,10 @@ func (bsnet *impl) SendMessage(
 	p peer.ID,
 	outgoing bsmsg.BitSwapMessage) error {
 
-	s, err := bsnet.newStreamToPeer(ctx, p)
+	tctx, cancel := context.WithTimeout(ctx, connectTimeout)
+	defer cancel()
```

Comment:
Taken from:
https://github.com/ipfs/go-bitswap/blob/a28f6eb5e764dfc5b05a57cb24181a57a007b686/network/ipfs_impl.go#L106-L116

Reply:
To clarify, I'm not setting the timeout _inside_ `newStreamToPeer` or as an argument (like `msgToStream` does) because the current pattern in `Connect` is embedding the timeout externally in the context passed to `newStreamToPeer`. But I can change both if we want to go in another direction.

Reply:
This is probably fine for now.